cpu/atmega2560/periph/spi.c

```diff
@@ -0,0 +1,235 @@
+/*
+* Copyright (C) 2015 Daniel Amkaer Sorensen
+*
+* This file is subject to the terms and conditions of the GNU Lesser General
+* Public License v2.1. See the file LICENSE in the top level directory for more
+* details.
+*/
+
+/**
+ * @ingroup     cpu_atmega2560
+ * @{
+ *
+ * @file
+ * @brief       Low-level SPI driver implementation
+ *
+ * @author      Daniel Amkaer Sorensen <daniel.amkaer@gmail.com>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "mutex.h"
+#include "sched.h"
+#include "thread.h"
+
+#include "periph/spi.h"
+#include "periph_conf.h"
+
+#if SPI_NUMOF
+
+/**
+ * @brief Each SPI device has to store one callback.
+ */
+typedef struct {
+    char (*rx_cb)(char data);
+} spi_state_t;
+
+static spi_state_t config[SPI_NUMOF];
+
+/**
+ * @brief Array holding one pre-initialized mutex for each SPI device
+ */
+static mutex_t locks[] =  {
+#if SPI_0_EN
+    [SPI_0] = MUTEX_INIT,
+#endif
+#if SPI_1_EN
+    [SPI_1] = MUTEX_INIT,
+#endif
+};
+
+int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
+{
+    uint8_t _spcr = (1<<SPI_0_MSTR)|(1<<SPI_0_SPE);
+
+    // Assume F_CPU = 16.000.000 Hz
+    switch (speed) {
+        case SPI_SPEED_100KHZ:
+            // Actual rate = 125.000 Hz
+            _spcr |= (1<<SPI_0_SPR1)|(1<<SPI_0_SPR0);
+            break;
+        case SPI_SPEED_400KHZ:
+            // Actual rate = 250.000 Hz
+            _spcr |= (1<<SPI_0_SPR1);
+            break;
+        case SPI_SPEED_1MHZ:
+            // Actual rate = 1.000.000 Hz
+            _spcr |= (1<<SPI_0_SPR0);
+            break;
+        case SPI_SPEED_5MHZ:
+            // Actual rate = 4.000.000 Hz
+            _spcr |= 0;
```

Comment:
Line without effect? Do you mind removing it and add a comment stating 4MHz is default?